### ISSUE TITLE
Suggestion: Add :set ft=language according to the header of the text

### ISSUE THREAD

**Issue by rafaelmuynarsk** [glacambre/firenvim#342]
When I open a piece of code on the browser with firenvim I always have to set the correct filetype with `:set ft=language-I-am-using` so I can see its highlight syntax... I'm not yet sure, but, is that the correct behavior? Is there a way to avoid setting up the filetype all the times that firenvim is launched?

If not, I have a suggestion... Couldn't it work exactly like on a terminal? Let's say that I save a bash file with the name `file` without the extension `.sh` on it... if I enter this file, I won't see any syntax highlight on it. However, once I put a header on the file like `#!/bin/bash` it'll open with the right syntax... For me it'd be great if firenvim could work like that as well.

**Comment by alerque** [comment]
Vim's file type detection relies heavily on filenames. Your example of shell scripts is a bit of an outlier because it does some snooping on file contents to disambiguate some file extensions (in your example, the case of no extension), and the hash-bang line is a pretty easy tip off. The vast majority of filetypes in vim won't be able to do any meaningful "peaking" at contents.

That's a problem for Firenvim because we don't have filenames to go off of. Also I think the vast majority of the time textareas on the web are going to be prose, not code. Your usage may vary, but the default file type of text is probably a good call.

The typical current solution is to set the file type defaults on a per site basis. For example I have all Firenvim edits on Github and Gitlab defaulting to Makdown, but on my self-hosted Gitlab site defaulting to Pandoc:

```vim
autocmd BufEnter github.com_*.txt set filetype=markdown
autocmd BufEnter gitlab.com_*.txt set filetype=markdown
autocmd BufEnter gitlab.alerque.com_*.txt set filetype=pandoc
```

You could do something similar for sites where you typically edit code in a certain language.

Eventually when there is a way to do this from the local settings configuration (already in the works) you could combine this with CSS selectors and some sites will probably have tips about the code language in CSS classes, so you could setup Firenvim with separate triggers based on each language's unique properties on the host site.

**Comment by glacambre** [comment]
One reason content-based syntax highlighting currently doesn't work is that Firenvim first asks neovim to open a buffer and only after that places the page's content into the buffer. For some reason I thought this would be hard to fix but I just thought of a potentially good solution.

But @alerque is right - most of the time, NeoVim won't be able to find out what the filetype of the buffer should be and you'll have to help it.

**Comment by rafaelmuynarsk** [comment]
Thanks for the fast responses!

@alerque The **autocmd** solves most of my problems but still there are some situations where the address doesn't define the language I'm using (Like when I'm editing code on Node-RED, it doesn't change the url address while I change between nodes)...

However, sorry if I'm going out of topic on this issue... You said you use Firenvim on Gitlab and GitHub, I didn't manage to make it edit texts on both of them. Gitlab opens Firevim with an empty file and GitHub has no response... Are there any extra steps to make it work? Should I open a new issue with the details of my case?

**Comment by glacambre** [comment]
This is fixed on master. I'll probably publish a new release tomorrow.
